src/include/firebird/FirebirdInterface.idl

```diff
@@ -1876,3 +1894,26 @@ interface ProfilerStats : Versioned
 {
 	uint64 getElapsedTicks();
 }
+
+// Extendable replacement for struct PerformanceInfo
+
+interface PerformanceCounters : Versioned
+{
+	uint getCount();
+	uint getVectorCapacity();
+
+	uint getId(uint index);
+	const string getName(uint index);
+	const int64* getCounterVector(uint index);
+}
+
+interface PerformanceStats : Versioned
+{
+	uint64 getElapsedTime();	// in milliseconds
+	uint64 getFetchedRecords();
+
+	PerformanceCounters getGlobalCounters();
```

Comment:
If you claim expandability, this must be one function with enum parameter: `cGlobal`, `cPage`, `cRelation`.

Reply:
We can easily add more functions to extend the interface. But I may agree a single function would be simpler.